hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/catalyst/catalog/HoodieCatalogTable.scala

```diff
@@ -164,12 +164,20 @@ class HoodieCatalogTable(val spark: SparkSession, val table: CatalogTable) exten
     val properties = new Properties()
     properties.putAll(tableConfigs.asJava)
 
-    HoodieTableMetaClient.withPropertyBuilder()
-      .fromProperties(properties)
-      .setTableName(table.identifier.table)
-      .setTableCreateSchema(SchemaConverters.toAvroType(finalSchema).toString())
-      .setPartitionFields(table.partitionColumnNames.mkString(","))
-      .initTable(hadoopConf, tableLocation)
+    if (hoodieTableExists) {
+      // just persist hoodie.table.create.schema
+      HoodieTableMetaClient.withPropertyBuilder()
+        .fromProperties(properties)
+        .setTableCreateSchema(SchemaConverters.toAvroType(finalSchema).toString())
```

Comment:
Hi @YannByron

Recently, we found an Avro schema issue which is caused by the wrong record name (detail here: https://github.com/apache/hudi/issues/7284).

May I ask if this line could cause the same problem? If so, we can discuss how to fix it in PR: https://github.com/apache/hudi/pull/7297

Reply:
Just tried, it will rewrite the record name in `hoodie.table.create.schema` as `topLevelRecord` in `hoodie.properties`.